### <span style="color:orange">Import useful modules</span>

In [26]:
import numpy as np
from scipy import constants as sc

In [33]:
def sciprint(a):
    print(f"{a:0.2g}")

### Critical density

$$
\boxed{
n_C = \frac{m_0 \epsilon_0 \omega^2}{e^2}
}
$$


#### Radiowaves refraction on the ionisphere

In [49]:
f = 5*1e6
w = 2*sc.pi*f

n_c = sc.m_e * sc.epsilon_0 * w**2 / sc.e**2
sciprint(n_c)

3.1e+11


#### Laser-Plasma

In [42]:
sciprint(sc.m_e * sc.epsilon_0 * (2*sc.pi * sc.c)**2 / sc.e**2 *1e6)

1.1e+21


$$
\boxed{
n_C \text{ [cm$^{-3}$]} = 1.1 \times 10^{21} \left( \frac{1}{\lambda \text{ [µm]}} \right)^2
}
$$